Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/DccValidationRepository.kt

```diff
@@ -0,0 +1,32 @@
+package de.rki.coronawarnapp.covidcertificate.validation.core
+
+import de.rki.coronawarnapp.covidcertificate.validation.core.country.DccCountry
+import de.rki.coronawarnapp.covidcertificate.validation.core.country.DccCountryRepository
+import de.rki.coronawarnapp.covidcertificate.validation.core.rule.DccValidationRuleRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DccValidationRepository @Inject constructor(
+    private val dccCountryRepository: DccCountryRepository,
+    private val validationRulesRepository: DccValidationRuleRepository,
+) {
+
+    val dccCountries: Flow<List<DccCountry>>
+        get() = dccCountryRepository.dccCountries
+
+    /**
+     * The UI calls this before entering the validation flow.
+     * Either we have a cached valid data to work with, or this throws an error for the UI to display.
+     */
+    suspend fun refresh() {
+        dccCountryRepository.refresh()
+        validationRulesRepository.refresh()
+    }
+
+    suspend fun clear() {
+        dccCountryRepository.clear()
+        validationRulesRepository.clear()
+    }
+}
```

Comment:
Created this as common API for the UI to work with.
After @chiljamgossow  is further along with the validation logic and @chris-cwa with the rule retrieval, we can evaluate whether the current abstraction still makes sense. For now this allows more people to work in parallel.
Removal will depend a bit on where data is accessed by injecting the repos and where data is passed via function argument.